aspects/schema_test.go

```diff
@@ -282,7 +282,7 @@ func (*schemaSuite) TestMapWithUnmetValuesConstraint(c *C) {
 	c.Assert(err, IsNil)
 
 	err = schema.Validate(input)
-	c.Assert(err, ErrorMatches, "cannot validate string: json: cannot unmarshal object into Go value of type string")
+	c.Assert(err, ErrorMatches, `cannot accept element in "snaps.foo": cannot parse string: json: cannot unmarshal object into Go value of type string`)
```

Comment:
not strictly for this PR, but I think we should change the code to avoid reporting directly json package errors for these situations, as we expect values to come from the API they should be usually wellformed json but maybe not of the right type: so maybe many places should do checks this way:

```
var v interface{}
... json.Unmarshal(raw, &v)
...
if _, ok := v.(string); !ok {
   return ... our own error ...
}
```

Reply:
True, this is still lacking. I've added a task to go over this as a follow-up